src/RemoteInboxNotifications/EvaluationLogger.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+/**
+ * Class EvaluationLogger
+ *
+ * @package Automattic\WooCommerce\Admin\RemoteInboxNotifications
+ */
+class EvaluationLogger {
+	/**
+	 * Slug of the spec.
+	 *
+	 * @var string
+	 */
+	private $slug;
+
+	/**
+	 * Results of rules in the given spec.
+	 *
+	 * @var array
+	 */
+	private $results = array();
+
+	/**
+	 * Logger class to use.
+	 *
+	 * @var WC_Logger_Interface|null
+	 */
+	private $logger;
+
+	/**
+	 * EvaluationLogger constructor.
+	 *
+	 * @param string                   $slug Slug of a spec that is being evaluated.
+	 * @param WC_Logger_Interface|null $logger Logger class to use.
+	 */
+	public function __construct( $slug, \WC_Logger_Interface $logger = null ) {
+		$this->slug = $slug;
+		if ( null === $logger ) {
+			$logger = wc_get_logger();
+		}
+
+		$this->logger = $logger;
+	}
+
+	/**
+	 * Add evaluation result of a rule.
+	 *
+	 * @param string  $rule_type name of the rule being tested.
+	 * @param boolean $result result of a given rule.
+	 */
+	public function add_result( $rule_type, $result ) {
+		array_push(
+			$this->results,
+			array(
+				'rule'   => $rule_type,
+				'result' => $result ? 'passed' : 'failed',
+			)
+		);
+	}
+
+	/**
+	 * Log the results.
+	 */
+	public function log() {
+		foreach ( $this->results as $result ) {
+			$this->logger->debug(
+				"[{$this->slug}] {$result['rule']}: {$result['result']}",
+				array( 'source' => 'remote-inbox-notifications' )
```

Comment:
I think the `source` here should be variable since this and rule logic is planned to be separated out from the remote inbox.

Reply:
@joshuatf Good catch. I did not factor that this can be used in other classes now.

Reply:
@joshuatf Sorry for the delay in response.

Updated the following changes in https://github.com/woocommerce/woocommerce-admin/pull/7194/commits/0bfb2520cdae4b37b794ebfb56d589d061716416

- Use a more generic constant name.
- source should be variable.
- Moved initialization of `EvaluationLogger` to `RuleEvaluator` class.